WordPress/Classes/ViewRelated/Blog/Blog Dashboard/ViewModel/BlogDashboardViewModel.swift

```diff
@@ -0,0 +1,79 @@
+import Foundation
+import UIKit
+import CoreData
+
+typealias DashboardSnapshot = NSDiffableDataSourceSnapshot<DashboardCardSection, DashboardCardModel>
+typealias DashboardDataSource = UICollectionViewDiffableDataSource<DashboardCardSection, DashboardCardModel>
+
+class BlogDashboardViewModel {
+    private weak var viewController: BlogDashboardViewController?
+
+    enum Section: CaseIterable {
+        case quickLinks
+        case posts
+    }
+
+    // FIXME: temporary placeholder
+    private let quickLinks = ["Quick Links"]
+    private let posts = ["Posts"]
+
+    typealias QuickLinksHostCell = HostCollectionViewCell<QuickLinksView>
+
+    private let managedObjectContext: NSManagedObjectContext
+    private let blog: Blog
+
+    private lazy var service: BlogDashboardService = {
+        return BlogDashboardService(managedObjectContext: managedObjectContext)
+    }()
+
+    private lazy var dataSource: DashboardDataSource? = {
+        guard let viewController = viewController else {
+            return nil
+        }
+
+        return DashboardDataSource(collectionView: viewController.collectionView) { [unowned self] collectionView, indexPath, identifier in
+
+            let cellType = identifier.id.cell
+            let cell = collectionView.dequeueReusableCell(withReuseIdentifier: cellType.defaultReuseID, for: indexPath)
+            (cell as? BlogDashboardCardConfigurable)?.configure(blog: blog, viewController: viewController, dataModel: identifier.cellViewModel)
+            return cell
```

Comment:
```suggestion
            guard let cell = collectionView.dequeueReusableCell(withReuseIdentifier: cellType.defaultReuseID, for: indexPath) as? BlogDashboardCardConfigurable else {
                 return UICollectionViewCell()
            }
            cell.configure(blog: blog, viewController: viewController, dataModel: identifier.cellViewModel)
            return cell
```

Reply:
One thing I'd like to avoid is returning a `UICollectionViewCell`. This shouldn't happen, for each registered card you need to provide a cell.

```suggestion
            let cell = collectionView.dequeueReusableCell(withReuseIdentifier: cellType.defaultReuseID, for: indexPath) as! BlogDashboardCardConfigurable
            cell.configure(blog: blog, viewController: viewController, dataModel: identifier.cellViewModel)
            return cell
```

What do you think of this approach?

Reply:
Sounds good -- I think we need cast to `DashboardCollectionViewCell` though. Otherwise we'll get a `Cannot convert return expression of type 'BlogDashboardCardConfigurable' to return type 'UICollectionViewCell?'` error

Reply:
Changed in 5c1f3cdee7b61528d87076c439b2d8aedb1d77c7 (ended up using `if let`), let me know wdyt.

Reply:
👍